wasm/src/lib.rs

```diff
@@ -0,0 +1,463 @@
+		serde_wasm_bindgen::from_value(sk_vec_js.clone())
+			.map_err(|_| JsError::new("could not decode secret key"))?;
+	let secret_key: [u8; 32] = sk_bytes.try_into().unwrap();
+
+	let ciphertext_vec: Vec<u8> =
+		serde_wasm_bindgen::from_value(ciphertext_js.clone())
+			.map_err(|_| JsError::new("could not decode ciphertext"))?;
+	let ciphertext_bytes: &[u8] = ciphertext_vec.as_slice();
+	let ciphertext: TLECiphertext<TinyBLS377> =
+		TLECiphertext::deserialize_compressed(ciphertext_bytes)
+			.map_err(|_| JsError::new("Could not deserialize ciphertext"))?;
+
+	let aes_ciphertext: AESOutput = AESOutput::deserialize_compressed(
+		&mut &ciphertext.body[..],
+	)
+	.unwrap();
+
+	let result =
+		AESGCMStreamCipherProvider::decrypt(aes_ciphertext, secret_key)
+			.map_err(|_| JsError::new("Message decryption failed"))?;
+
+	let plaintext: String = String::from_utf8(result).map_err(|_| {
+		JsError::new("Plaintext could not be converted to a string")
+	})?;
+
+	serde_wasm_bindgen::to_value(&plaintext)
+		.map_err(|_| JsError::new("plaintext conversion has failed"))
+}
+
+/// Logging struct, useful for testing and debugging
+#[wasm_bindgen]
+extern "C" {
+	#[wasm_bindgen(js_namespace = console)]
+	fn log(s: &str);
+}
+
+/// Struct for testing that allows for the serialization of the double public
+/// key type
+#[derive(
+	Serialize, CanonicalSerialize, CanonicalDeserialize, Deserialize, Debug,
+)]
+pub struct KeyChain {
+	#[serde(with = "BigArray")]
+	pub double_public: [u8; 144],
+
+	pub sk: [u8; 32],
+}
+
+/// build an encoded commitment for use in timelock encryption and sig
+/// verification
+#[wasm_bindgen]
+pub fn build_encoded_commitment(
+	block_number_js: JsValue,
+	validator_set_id_js: JsValue,
+) -> Result<JsValue, JsError> {
+	let block_number: u32 = serde_wasm_bindgen::from_value(
+		block_number_js.clone(),
+	)
+	.map_err(|_| JsError::new("could not decode a u32 from the input"))?;
+	let validator_set_id: u64 = serde_wasm_bindgen::from_value(
+		validator_set_id_js.clone(),
+	)
+	.map_err(|_| JsError::new("could not decode a u32 from the input"))?;
+	let payload =
+		Payload::from_single_entry(known_payloads::ETF_SIGNATURE, Vec::new());
+	let commitment = Commitment { payload, block_number, validator_set_id };
+	let encoded = commitment.encode();
+	serde_wasm_bindgen::to_value(&encoded).map_err(|_| {
+		JsError::new("could not convert the encoded commitment to json")
+	})
+}
+
+/// This function is used purely for testing purposes.
+/// It takes in a seed and generates a secret key and public params.
+#[wasm_bindgen]
+pub fn generate_keys(seed: JsValue) -> Result<JsValue, JsError> {
+    let seed_vec: Vec<u8> = serde_wasm_bindgen::from_value(seed)
+        .map_err(|_| JsError::new("Could not convert seed to string"))?;
+    let seed_vec = seed_vec.as_slice();
+
+	let mut hasher = sha2::Sha256::default();
+	hasher.update(&seed_vec);
+	let hash = hasher.finalize();
+    let seed_hash: [u8; 32] = hash.try_into().unwrap();
```

Comment:
Can we use the infallible `into` instead?
```suggestion
    let seed_hash: [u8; 32] = hash.into();
```